tests/test_Timeout.cc

```diff
@@ -0,0 +1,144 @@
+/**
+ * \file   test_Timeout.cc
+ * \author Lars Froehlich
+ * \date   Created on November 28, 2022
+ * \brief  Test suite for the Timeout class.
+ *
+ * \copyright Copyright 2022 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include <cmath>
+#include <stdexcept>
+
+#include <gul14/catch.h>
+
+#include "taskolib/Timeout.h"
+
+using namespace std::literals;
+using namespace task;
+using namespace Catch::Matchers;
+
+TEST_CASE("Timeout: Default constructor", "[Timeout]")
+{
+    static_assert(std::is_default_constructible<Timeout>::value,
+        "Timeout is_default_constructible");
+
+    Timeout t;
+    REQUIRE(t.is_infinite());
+}
+
+TEST_CASE("Timeout: Construction from chrono duration", "[Timeout]")
+{
+    REQUIRE_THROWS_AS(Timeout{ -1ms }, Error);
+    REQUIRE_THROWS_AS(Timeout{ -10h }, Error);
+    REQUIRE(Timeout{ 0s }.is_finite());
+    REQUIRE(Timeout{ 1s }.is_finite());
+    REQUIRE(Timeout{ 24h }.is_finite());
+    REQUIRE(Timeout{ std::chrono::duration<double>{ 1.0e100 } }.is_infinite());
+}
+
+TEST_CASE("Timeout: Construction from floating-point value", "[Timeout]")
+{
+    REQUIRE_THROWS_AS(Timeout{ -10.0 }, Error);
+    REQUIRE(Timeout{ 0.0f }.is_finite());
+    REQUIRE(Timeout{ 1.0 }.is_finite());
+    REQUIRE(Timeout{ 24.0f * 60.0f * 60.0f }.is_finite());
+    REQUIRE(Timeout{ 1e100 }.is_infinite());
+    REQUIRE(Timeout{ NAN }.is_infinite());
```

Comment:
Should that not rather `throw` :thinking: 
Or be zero.
When a NAN comes it it is almost certain some error/bug, and when that is used as infinity, that makes detection rarther hard.

Reply:
Good point. I'll change the behavior of the constructor.

Reply:
Done.